ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -170,6 +171,9 @@ class ( Show (ChainDepState   p)
   -- | We require that protocols support a @k@ security parameter
   protocolSecurityParam :: ConsensusConfig p -> SecurityParam
```

Comment:
Hmm. This is an awkward juxtaposition :/
```
  protocolSecurityParam                 :: ConsensusConfig p -> SecurityParam
  protocolSecurityParamConsistencyCheck :: ConsensusConfig p -> Maybe (NonEmpty SecurityParam)
```

I'm not sure what to do about it. My initial idea is to make the check specific to the hardfork combinator somehow? 🤔 Since that's where block types collide. (Ie don't put it in our of our fully abstract classes like `ConsensusProtocol` here.

Or perhaps just changing the return type to `Maybe StartupWarning` would suffiice to avoid the awkwardness? Still kind of feels like it belongs in a different class

Reply:
removed this after alex suggested an entirely separate `BlockSupportsSanityCheck` class